### PR TITLE
UIRR-6

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose
[UIRR-6](https://folio-org.atlassian.net/browse/UIRR-6) - Display patron details on patron's barcode Scan.

**Please note that the above jira is a subtask of a [story ](https://folio-org.atlassian.net/browse/UIRR-4)- hence few lines may seem insignificant. They will be improved in the scope of further sub tasks.**

Just as an additional information -
7/22 files - are mock files
1/22 file - packages.json
1/22 - translation file
1/22 - changelog.md
1/22 - yarn.lock
2/22 - index.js file from various folders

**overall, there are 9 files for actual feature implementation and unit tests together.**

## Approach
The screen should offer an input field to enter barcode manually or back fill it from patron look up.
On clicking enter button, i.e., on scanning patron barcode, patron details needs to be displayed.
Incase profile picture feature is enabled for the tenant and scanned patron has image attached to his profile(otherwise a thumbnail placeholder image), his profile picture should be displayed as well.

## Screencast
chrome_V5lCsppVFe

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/ScanPatron/ScanPatron.js [hunk 1]
```diff
@@ -1,7 +1,124 @@
...
+const ScanPatron = ({ mutator, resources }) => {
+  const isUserProfilePicConfigEnabledForTenant = get(resources, 'userProfilePicConfig.records[0].enabled');
+  const [scannedPatronDetails, setScannedPatronDetails] = useState();
+
+  const handleScanPatron = async (barcode) => {
+    if (barcode) {
+      const query = `barcode==${barcode}`;
+      const patron = await mutator.patrons.GET({ params: { query } });
+
+      if (patron.length) {
```
Comment (manvendra-s-rathore) [inline src/ScanPatron/ScanPatron.js:26]: can patron variable be undefined ? Is optional chaining needed here, please have a look.

File: test/jest/__mock__/stripesComponents.mock.js [hunk 2]
```diff
@@ -147,4 +165,15 @@ jest.mock('@folio/stripes/components', () => ({
       {children}
     </div>
   )),
+  TextField: jest.fn((props) => {
+    console.log('props ', props);
```
Comment (manvendra-s-rathore) [inline test/jest/__mock__/stripesComponents.mock.js:169]: remove this console log.

File: src/ScanPatron/ScanPatron.js [hunk 3]
```diff
@@ -1,7 +1,124 @@
...
+    <Paneset>
+      <Pane
+        id="reading-room"
+        paneTitle={<FormattedMessage id="ui-reading-room.scanPatronCard" />}
+        centerContent
+        defaultWidth="fill"
+        footer={footer}
+      >
+        <ScanForm
+          onSubmit={() => {}}
```
Comment (Terala-Priyanka) [inline src/ScanPatron/ScanPatron.js:81]: Please note that the above jira is a subtask of a [story ](https://folio-org.atlassian.net/browse/UIRR-4)- hence few lines may seem insignificant. They will be improved in the scope of further sub tasks.

File: test/jest/__mock__/stripesComponents.mock.js (hunk 2 above)
Comment (Terala-Priyanka) [reply test/jest/__mock__/stripesComponents.mock.js]: done

File: src/ScanPatron/ScanPatron.js (hunk 1 above)
Comment (Terala-Priyanka) [reply src/ScanPatron/ScanPatron.js]: 1. added check for patron
2. this block of code will grow and hence I don't think optional chaining will be suitable here.

**Comment by Terala-Priyanka** [comment]
@folio-org/fe-tl-reviewers 
Could you please take time to review this PR?

**Review by manvendra-s-rathore — APPROVED** [review]

**Comment by Terala-Priyanka** [comment]
@folio-org/fe-tl-reviewers
Could you please take time to review this PR?

**Pull request merged**